### ISSUE TITLE
Feature request: Modify index.html in flight

### ISSUE THREAD

**Issue by patryk-smc** [szymmis/vite-express#58]
Hello,
first of all, I wanted to thank you for making this package. In the past, I spent countless hours wiring Express and Vite together and this is 100x better than what I got. 

I have a use case where I need to replace a placeholder set in index.html with user-specific data. This happens after authentication middleware. 

To illustrate:

My src/index.html looks like this:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React + TS</title>
    <!-- App data -->
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/client/main.tsx"></script>
  </body>
</html>
```

I'd like to replace `<!-- App data -->` with dynamically set `userId` like that:

```html
<script type="text/json" id="app-data">{"userId": "123"}</script>
```

Is this something that could be added? I would be happy to sponsor this feature.

**Comment by patryk-smc** [comment]
Figured it out on my own :)

**Comment by szymmis** [comment]
Hi @patryk-smc, sorry for late response. I think it is a great idea and it also solved problem described in #56. If you are still interested you can check out my idea that I've described in that issue, if not I'd be happy to se your solution to the problem.

Thanks for you input, and again, sorry for such a delay in response.
